### PR TITLE
Linked list tut

### PR DESCRIPTION
**Pull request by itsmenikhill**
## Change Summary

Complete linked list tutorial. 

## Checklist

**Ideally all of them should be checked. If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed.**

### General 

- [x ] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x ] Make sure you've followed the template format
- [ x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x ] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [ x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
// See https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr for example
```

</TabItem>

</Tabs>
````
- [ x] Make sure you've explained your code well (write comments above each line - not inline)
- [ x] Make sure you've written your explanation well and it is easy to understand for beginners
- [ x] Make sure you've included Time & Space Complexity Analysis
- [x ] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [x ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ x] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [x ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

**Comment by itsmenikhill** [comment]
@wingkwong hey can you please take a look

File: tutorials/basic-topics/linked-list.md [hunk 1]
```diff
@@ -1,10 +1,688 @@
 ---
 title: 'Linked List'
-description: ''
+description: 'A linked list is a linear data structure that consists of nodes. Each node stores some data as well as the address of the next node.'
 hide_table_of_contents: true
```
Comment (wingkwong) [inline tutorials/basic-topics/linked-list.md:4]: make it to false as you have different levels

File: tutorials/basic-topics/linked-list.md [hunk 2]
```diff
@@ -1,10 +1,688 @@
...
   - linked List
----
\ No newline at end of file
+  - data structure
+---
+
+<TutorialAuthors names="@itsmenikhill"/>
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
```
Comment (wingkwong) [inline tutorials/basic-topics/linked-list.md:15]: these are not necessary as I've imported them globally weeks ago.

File: tutorials/basic-topics/linked-list.md [hunk 3]
```diff
@@ -1,10 +1,688 @@
...
+* Address of next node
+
+Both of these items are wrapped together in a class:
+
+<Tabs
+    defaultValue="java"
+    values={[
+        {label: 'Java', value: 'java'},
+    ]}>
+<TabItem value="java">
```
Comment (wingkwong) [inline tutorials/basic-topics/linked-list.md:39]: Use

```
<Tabs>
<TabItem value="java" label="Java">
```

Apply to all.

File: tutorials/basic-topics/linked-list.md [hunk 4]
```diff
@@ -1,10 +1,688 @@
...
+  // move ptr to the next node
+  ptr = ptr.next;  
+}
+// return false if number not found
+return false;  
+``` 
+</TabItem>
+</Tabs>
+
+## Let's take a look at some Leetcode problems:
```
Comment (wingkwong) [inline tutorials/basic-topics/linked-list.md:446]: Before this line you've already shown some Leetcode problems.

File: tutorials/basic-topics/linked-list.md [hunk 5]
```diff
@@ -1,10 +1,688 @@
...
+
+**Leetcode 19. Remove Nth Node From End of List**
+
+Given the head of a linked list, remove the nth node from the end of the list and return its head. Given Linked list: $[1] -> [2] -> [3] -> [4] -> [5]$ for $n=2$ change the list to $[1] -> [2] -> [3] ->     [5]$.
```
Comment (wingkwong) [inline tutorials/basic-topics/linked-list.md:450]: looks weird in the app.

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. at the beginning, you refer $Head$ but you refer $head$ in the middle. Please align the case.
2. for each leetcode example, change the format like below
```
### Example: [0303 - Range Sum Query - Immutable](https://leetcode.com/problems/range-sum-query-immutable/)
```

File: tutorials/basic-topics/linked-list.md [hunk 6]
```diff
@@ -1,10 +1,688 @@
...
+**234. Palindrome Linked List**
+
+Given the $head$ of a singly linked list, return $true$ if it is a $palindrome$ or $false$ otherwise. 
+
+Example: 
+
+
+![image](https://user-images.githubusercontent.com/44477975/196033584-ca91cbc9-0126-40d2-8788-4fbe3b50d55a.png)
+
+$Output: true$
```
[the remark this answers is not in the document]
Comment (itsmenikhill) [reply tutorials/basic-topics/linked-list.md:524]: you want me to remove the image also?

File: tutorials/basic-topics/linked-list.md (hunk 4 above)
Comment (itsmenikhill) [reply tutorials/basic-topics/linked-list.md:446]: ok so should i change it to just "Problems" then?

File: tutorials/basic-topics/linked-list.md (hunk 5 above)
Comment (itsmenikhill) [reply tutorials/basic-topics/linked-list.md:450]: just this one or everywhere in the app. I can replace it with an image.

Comment (wingkwong) [reply tutorials/basic-topics/linked-list.md:450]: Use the following

```
Given the head of a linked list, remove the nth node from the end of the list and return its head. Given Linked list: $[1]$ -> $[2]$ -> $[3]$ -> $[4]$ -> $[5]$ for $n = 2$ change the list to $[1]$ -> $[2]$ -> $[3]$ -> $[5]$.
```

**Comment by itsmenikhill** [comment]
Hi, I have made the necessary changes. Kindly review.

**Review by wingkwong — APPROVED** [review]
To save my time, I've cleaned it up.

**Pull request merged**